### PR TITLE
Review 19 Jan 2021

### PR DESCRIPTION
**Pull request by fangyi-zhou**

### CODE REVIEW

File: lib/lib.ml [hunk 1]
```diff
@@ -40,9 +40,8 @@ let validate_protocols_exn (ast : scr_module) ~verbose : unit =
   let g_types =
     List.map ~f:(fun p -> (Gtype.of_protocol p, p.value.roles)) protocols
   in
-  let g_types =
-    List.map ~f:(fun (g, roles) -> (Gtype.normalise g, roles)) g_types
-  in
+  (* let g_types = List.map ~f:(fun (g, roles) -> (Gtype.normalise g, roles))
```
Comment (fferreira) [inline lib/lib.ml:43]: if don't normalise the g then we could remove it. Or was it temporary?

File: lib/lib.ml [hunk 2]
```diff
@@ -68,7 +67,7 @@ let validate_nested_protocols (ast : scr_module) ~verbose =
   let ast = Protocol.rename_nested_protocols ast in
   show ~f:show_scr_module ~sep:"\n---------\n\n" ast ;
   let global_t = Gtype.global_t_of_module ast in
-  let global_t = Gtype.normalise_global_t global_t in
+  (* let global_t = Gtype.normalise_global_t global_t in *)
```
Comment (fferreira) [inline lib/lib.ml:70]: same, do we need to keep this?

File: test/test_gtype.ml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+(*
```
Comment (fferreira) [inline test/test_gtype.ml:1]: This file is all commented out, do we really need it?

**Review by fferreira — COMMENTED** [review]
The cram test part I have never used so I don't understand much. I have one question though: it seems we are comparing the output of code generation to stored correct outputs. Isn't this gooing to lead to maintanability problems for small changes in whitespace or things like that? There are already renumbering of states in the committed "correct" code.

File: test/test_gtype.ml (hunk 3 above)
Comment (fangyi-zhou) [reply test/test_gtype.ml:1]: https://github.com/nuscr/nuscr/commit/b7ff56477c7b1d58553e702f467ca61b6cff5e6a

File: lib/lib.ml (hunk 1 above)
Comment (fangyi-zhou) [reply lib/lib.ml:43]: https://github.com/nuscr/nuscr/issues/130

File: lib/lib.ml (hunk 2 above)
Comment (fangyi-zhou) [reply lib/lib.ml:70]: https://github.com/nuscr/nuscr/issues/130

**Review by fferreira — APPROVED** [review]

**Pull request merged**